book.rb

```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+    rental.book = self
+  end
```

Comment:
- [ ] I think the way we're creating the rental here is not correct. I explain myself. We are creating a new rental of the book, which means that passing a rental as an argument does not make much sense. Instead, we want to create a new rental with the classes involved in the many-to-many relationship: Book, Person, and Rental.

We can achieve this by passing a person and a date as arguments and then using them to instantiate a new rental. It looks similar to this:

```
def add_rental(person, date)
  Rental.new(date, self, person)
end
```

Then Rental class will create all of the relationships correctly.